google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -165,6 +128,48 @@ public static void addSupport(@NotNull AppEngineFlexibleFacet facet,
     setupDeploymentRunConfiguration(facet.getModule());
   }
 
+  /**
+   * Generates an app.yaml configuration file in the src/main/appengine directory. If an app.yaml
+   * already exists it will not overwrite the file.
+   */
+  private static void generateAppYaml(@NotNull AppEngineFlexibleFacet facet,
+      @NotNull ModifiableRootModel rootModel) {
+    Project project = facet.getModule().getProject();
+    FileTemplate appYamlTemplate = FileTemplateManager.getInstance(project)
+        .getInternalTemplate(AppEngineTemplateGroupDescriptorFactory.APP_YAML_TEMPLATE);
+
+    try {
+      VirtualFile virtualFile = rootModel.getContentRoots()[0].findFileByRelativePath("src/main");
+
+      if (virtualFile != null) {
+        PsiDirectory directory = PsiManager.getInstance(project)
+            .findDirectory(virtualFile);
+
+        if (directory != null) {
+          PsiDirectory appEngineDirectory;
+          try {
+            directory.checkCreateSubdirectory(CONFIG_DIR_NAME);
+            appEngineDirectory = directory.createSubdirectory(CONFIG_DIR_NAME);
+          } catch (IncorrectOperationException ioe) {
+            // checkCreateSubdirectory threw an exception suggesting that the directory may already
+            // exist. Skip creating the directory and attempt to write file.
+            appEngineDirectory = directory.findSubdirectory(CONFIG_DIR_NAME);
+          }
+
+          if (appEngineDirectory != null
+              && FileTemplateUtil.canCreateFromTemplate(
+                  new PsiDirectory[]{appEngineDirectory}, appYamlTemplate)) {
+            FileTemplateUtil.createFromTemplate(appYamlTemplate, "app.yaml",
+                FileTemplateManager.getInstance(project).getDefaultProperties(),
+                appEngineDirectory);
+          }
+        }
+      }
+    } catch (Exception e) {
```

Comment:
on second, probably fail so we can know about it. If the directory was located / created and creating the file from the template fails then this really is unexpected. i'll refactor.